101-data-factory-blob-to-sql-copy-stored-proc/azuredeploy.json

```diff
@@ -0,0 +1,246 @@
+      "metadata": {
+        "description": "Name of Azure SQL Database in Azure SQL server copying data to."
+      }
+    },
+    "sqlUserID": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for access to Azure SQL Server."
+      }
+    },
+    "sqlPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the user to Azure SQL Server."
+      }
+    },
+    "sqlTargetTable": {
+      "type": "string",
+      "metadata": {
+        "description": "Table in the Azure SQL Database that will hold the copied data."
+      }
+    },
+    "sqlWriterTableType": {
+      "type": "string",
+      "metadata": {
+        "description": "Specify a table type name to be used in the stored procedure. Copy activity makes the data being moved available in a temp table with this table type."
+      }
+    },
+    "sqlWriterStoredProcedureName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the stored procedure that upserts (updates/inserts) data into the target table."
+      }
+    }
+  },
+  "variables": {
+    "dataFactoryName": "[concat('CopyFromAzureBlobToAzureSQLDbSproc', uniqueString(resourceGroup().id))]",
+    "storageLinkedServiceName": "StorageLinkedService",
+    "sqlLinkedServiceName": "SqlLinkedService",
+    "storageDataset": "StorageDataset",
+    "intermediateDataset": "IntermediateDataset",
+    "sqlDataset": "SqlDataset",
+    "pipelineName": "BlobtoSqlDbCopyPipelineSproc"
+  },
+  "resources": [
+    {
+      "name": "[variables('dataFactoryName')]",
+      "apiVersion": "2015-10-01",
+      "type": "Microsoft.DataFactory/datafactories",
+      "location": "West US",
+      "resources": [
+        {
+          "type": "linkedservices",
+          "name": "[variables('storageLinkedServiceName')]",
+          "apiVersion": "2015-10-01",
+          "dependsOn": [ "[variables('dataFactoryName')]" ],
+          "properties": {
+            "type": "AzureStorage",
+            "description": "Azure Storage Linked Service",
+            "typeProperties": {
+              "connectionString": "[concat('DefaultEndpointsProtocol=https;AccountName=',parameters('storageAccountName'),';AccountKey=',parameters('storageAccountKey'))]"
+            }
+          }
+        },
+        {
+          "type": "linkedservices",
+          "name": "[variables('sqlLinkedServiceName')]",
+          "apiVersion": "2015-10-01",
+          "dependsOn": [ "[variables('dataFactoryName')]" ],
+          "properties": {
+            "type": "AzureSqlDatabase",
+            "description": "Azure SQL linked service",
+            "typeProperties": {
+              "connectionString": "[concat('Data Source=tcp:', parameters('sqlServerName'), '.database.windows.net,1433;Initial Catalog=', parameters('sqlDatabaseName'), ';Integrated Security=False;User ID=', parameters('sqlUserId'), ';Password=', parameters('sqlPassword'), ';Connect Timeout=30;Encrypt=True')]"
```

Comment:
User specifies the Azure SQL 1) server name 2) database name 3) user 4) password in the parameter file.